cql3/functions/user_aggregate.hh

```diff
@@ -37,14 +37,14 @@ public:
     virtual sstring element_type() const override { return "aggregate"; }
     virtual std::ostream& describe(std::ostream& os) const override;
 
-    const scalar_function& sfunc() const {
-        return *_sfunc;
+    ::shared_ptr<scalar_function> sfunc() const {
```

Comment:
If you want to disambiguate shared_ptr, please use seastar::shared_ptr not ::shared_ptr which is confusing (at least for me).

I didn't understand from the commit message what is really the motivation with that change. Why isn't a bare pointer enough? A shared pointer is important if you need to keep this pointer for a long time and need to ensure the object lives during that time - but do you? I guess I'll see in a later patch.

Reply:
It looks like the `::shared_ptr` is very common in this area of code (`cql3/statements`), but I agree that `seastar::shared_ptr` is less confusing - I'll add it.

I did not give the exact motivation for this change in the commit message, but as you've probably seen, we do need the `shared_ptr` because we're actually copying it to reconstruct an affected UDA in the last commit of the series.